github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml

```diff
@@ -44,11 +44,78 @@ postsubmits:
           - "/usr/local/bin/runner.sh"
           - "/bin/bash"
           - "-c"
-          - >
+          - |
             cat $QUAY_PASSWORD | podman login --username $(<$QUAY_USER) --password-stdin quay.io &&
             ./publish.sh &&
-            echo "$(git tag --points-at HEAD | head -1)" > latest &&
-            gsutil cp ./latest gs://kubevirt-prow/release/kubevirt/kubevirtci/latest
+            # Gets git tag created by above publish.sh run
+            KUBEVIRTCI_TAG=$(git tag --points-at HEAD | head -1) &&
+            echo "$KUBEVIRTCI_TAG" > latest &&
+            gsutil cp ./latest gs://kubevirt-prow/release/kubevirt/kubevirtci/latest &&
+            SHORT_SHA=$(git rev-parse --short HEAD) &&
+            echo "$KUBEVIRTCI_TAG" > amd64-$SHORT_SHA &&
+            gsutil cp ./amd64-$SHORT_SHA gs://kubevirt-prow/release/kubevirt/kubevirtci/amd64-$SHORT_SHA
+          # docker-in-docker needs privileged mode
+          env:
+          - name: GIMME_GO_VERSION
+            value: "1.22.5"
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - mountPath: /dev
+            name: devices
+          resources:
+            requests:
+              memory: "29Gi"
+    - name: publish-kubevirtci-s390x
+      branches:
+      - main
+      always_run: true
+      annotations:
+        testgrid-create-test-group: "false"
+      decorate: true
+      decoration_config:
+        timeout: 2h
+      max_concurrency: 1
+      extra_refs:
+      - org: kubevirt
+        repo: project-infra
+        base_ref: main
+      labels:
+        preset-podman-in-container-enabled: "true"
+        preset-gcs-credentials: "true"
+        preset-kubevirtci-quay-credential: "true"
+      cluster: prow-s390x-workloads
+      spec:
+        volumes:
+        - hostPath:
+            path: /dev
+            type: Directory
+          name: devices
+        containers:
+        - image: quay.io/kubevirtci/golang:v20241014-80f340c
+          command:
+          - "/usr/local/bin/runner.sh"
+          - "/bin/bash"
+          - "-c"
+          - |
+            cat $QUAY_PASSWORD | podman login --username $(<$QUAY_USER) --password-stdin quay.io &&
+            SHORT_SHA=$(git rev-parse --short HEAD) &&
+            GCS_FILE_PATH=release/kubevirt/kubevirtci/amd64-$SHORT_SHA &&
```

Comment:
Why don't we just curl for the expected public path? then we could avoid the need for gcs credentials etc. here?

Reply:
You are right for this part. I've changed the code locally, but later realized that as we anyway need credentials as we are removing the file `        rm_gcs_file kubevirt-prow "$GCS_FILE_PATH"` below, probably we can leave this as it is.